website/src/pages/specification/20241027-flagsets.md

```diff
@@ -0,0 +1,200 @@
+- Single retrievers for multiple flag sets _(as proposed in https://github.com/thomaspoignant/go-feature-flag/issues/2314)_.
+
+_Even if out of scope for now, those are interesting options that we may want to implement later._
+
+## Proposed solution
+
+### Solution 1: 1 flag set per file
+
+In this solution we consider that we need at least one file per flag set.  
+All the flags retrieved by a `retriever` _(aka in the same file)_ will be associated to the same flag set.
+
+We can specify the flag set name in the configuration file.  
+As of today, we can still have multiple files for 1 flag set _(by specifying the same `flagSet` name in each file)_,
+and we will have the same mechanism as of today _(with flag overridden in case of flag name collision)_.
+
+#### Flags configuration file
+```yaml
+# config-file1.goff.yaml
+# Syntax used in this example is just for the sake of the example, it is not the final syntax.
+flagSet: flagset-teamA
+
+featureA-enabled:
+  variations:
+      enabled: true
+      disabled: false
+  defaultRule:
+    variation: enabled
+```
+```yaml
+# config-file2.goff.yaml
+flagSet: flagset-teamB
+
+featureA-enabled:
+  variations:
+      enabled: true
+      disabled: false
+  defaultRule:
+    variation: enabled
+```
+
+#### Relay-proxy configuration example
+```yaml
+# ...
+retrievers:
+  - kind: file
+    path: config-file1.goff.yaml
+  - kind: file
+    path: config-file2.goff.yaml
+
+authorizedKeys:
+  evaluation:
+    - apikey1 # owner: userID1
+    - apikey2 # owner: userID2
+  admin:
+    - apikey3
+```
+
+**PRO**
+- It is simple to configure a flag set by putting all the flags at the same place.
+- It is easy to understand which flags are part of a flag set.
+- It is easy to give ownership of a flag set to a team, by giving them access to the file.
+
+**CON**
+- If we want the same flag to be part of multiple flag sets, we need to duplicate the flag in multiple files.
+
+### Solution 2: specify the flag set in the retriever configuration
+
+In this solution we consider that we need at least one file per flag set.  
+All the flags retrieved by a `retriever` _(aka in the same file)_ will be associated to the same flag set.
+
+We associate the flag set to the retriever in the configuration,
+#### Flags configuration file
+```yaml
+# config-file1.goff.yaml
+featureA-enabled:
+  variations:
+      enabled: true
+      disabled: false
+  defaultRule:
+    variation: enabled
+```
+```yaml
+# config-file2.goff.yaml
+featureA-enabled:
+  variations:
+      enabled: true
+      disabled: false
+  defaultRule:
+    variation: enabled
+```
+
+#### Relay-proxy configuration example
+```yaml
+# ...
+retrievers:
+  - kind: file
+    path: config-file1.goff.yaml
+    flagSet: flagset-teamA
+  - kind: file
+    path: config-file2.goff.yaml
+    flagSet: flagset-teamB
+
+authorizedKeys:
+  evaluation:
+    - apikey1 # owner: userID1
+    - apikey2 # owner: userID2
+  admin:
+    - apikey3
```

Comment:
🤔 I am not sure how to configure the relation between the API keys and the flag sets, all proposals are welcome on this.